lora-phy/src/lib.rs

```diff
@@ -66,6 +72,20 @@ where
         Ok(lora)
     }
 
+    /// Build and return a new instance of the LoRa physical layer API to
+    /// control an initialized LoRa radio for LoRaWAN public or private network.
+    ///
+    /// In order to configure radio to use non-LoRaWAN networks, use
+    /// [`Self::with_syncword()`] which has `sync_word` argument.
+    pub async fn new(radio_kind: RK, enable_public_network: bool, delay: DLY) -> Result<Self, RadioError> {
+        let sync_word = if enable_public_network {
+            LORAWAN_PUBLIC_SYNCWORD
+        } else {
+            LORAWAN_PRIVATE_SYNCWORD
+        };
```

Comment:
So I guess this kind of violates our rules about protocol level stuff in the radio-driver, but it's maintained to preserve the API?

Reply:
Yeah, this is also defined in trait:
https://github.com/lora-rs/lora-rs/blob/5664d38c692da8574adc1dd353de29d307540ab4/lora-phy/src/mod_traits.rs#L36-L37